Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc

```diff
@@ -1169,10 +1174,12 @@ static long algorithm(dd4hep::Detector& /* description */, cms::DDParsingContext
               const double yy(radius * sin(phi));
               const double xx(radius * cos(phi));
               ++ptmCopy;
-              xilyLog.placeVolume(
-                  ilyPTMLog,
-                  ptmCopy,
-                  Transform3D(RotationZ(phi), Position(xx, yy, ily.vecIlyPTMZ[ilyPTM] - ilyLengthHalf)));
+              xilyLog.placeVolume(ilyPTMLog,
+                                  ptmCopy,
+                                  Transform3D(myrot(ns,
+                                                    std::string(ilyPTMLog.name()) + "_rot" + std::to_string(ptmCopy),
```

Comment:
@cvuosalo - this is an unnecessary performance bomb. A new object is returned after each +. That means that a new buffer is used each time. You are doing a ton of extra + operations - it is not efficient.

Reply:
I will remove these rotation names since they are not strictly necessary.